electrum/daemon.py

```diff
@@ -96,15 +101,29 @@ def request(config: SimpleConfig, endpoint, args=(), timeout=60):
         create_time = None
         try:
             with open(lockfile) as f:
-                (host, port), create_time = ast.literal_eval(f.read())
+                socktype, address, create_time = ast.literal_eval(f.read())
+                if socktype == 'unix':
+                    path = address
+                    (host, port) = "127.0.0.1", 0
+                    # We still need a host and port for e.g. HTTP Host header
+                elif socktype == 'tcp':
+                    (host, port) = address
+                else:
+                    assert False
```

Comment:
Please only use asserts for sanity checks or to clearly state your assumptions for readers, not for important logic.
Raise exceptions instead (guarded with if branches).
Asserts are not guaranteed to be there at runtime: https://docs.python.org/3/using/cmdline.html#envvar-PYTHONOPTIMIZE

```suggestion
                    raise Exception(f"unknown socktype={socktype!r}")
```

Reply:
This is a sanity check. It's not supposed to be possible, I'm not supposed to have random non-existent socktypes in my lockfile. As for the command-line option:
```
$ ./run_electrum --regtest daemon --rpcsock=x    
usage: run_electrum daemon [-h] [-d] [--rpcsock {unix,tcp,auto}]
                           [--rpcsockpath RPCSOCKETPATH]
                           [-f SERVERFINGERPRINT] [-1] [-s SERVER] [-p PROXY]
                           [--noonion] [--skipmerklecheck] [-v VERBOSITY]
                           [-V VERBOSITY_SHORTCUTS] [-D ELECTRUM_PATH] [-P]
                           [--testnet] [--regtest] [--simnet] [--signet] [-o]
run_electrum daemon: error: argument --rpcsock: invalid choice: 'x' (choose from 'unix', 'tcp', 'auto')
```

Reply:
The issue is that the user might set the config var directly, not via CLI option, in which case the possible values are unrestricted.
The same applies (and it's more apparent) to the other lines where you added an `assert False`.
`daemon.socktype` can have any value currently

Reply:
Right, I missed that. I changed it everywhere for consistency/simplicity, although it would arguably be more correct in a sense to keep the asserts for the lockfile.